test/200_fullchain-bot.test.ts

```diff
@@ -0,0 +1,314 @@
+const appPrice = 1000;
+const datasetPrice = 1_000_000;
+const workerpoolPrice = 1_000_000_000;
+
+let proxyAddress: string;
+let iexecPoco: IexecInterfaceNative;
+let iexecWrapper: IexecWrapper;
+let [appAddress, workerpoolAddress, datasetAddress]: string[] = [];
+let [
+    requester,
+    appProvider,
+    datasetProvider,
+    scheduler,
+    anyone,
+    worker1,
+    worker2,
+    worker3,
+    worker4,
+    worker5,
+]: SignerWithAddress[] = [];
+let ordersActors: OrdersActors;
+let ordersAssets: OrdersAssets;
+let ordersPrices: OrdersPrices;
+
+describe('Integration tests', function () {
+    beforeEach('Deploy', async () => {
+        // Deploy all contracts
+        proxyAddress = await loadHardhatFixtureDeployment();
+        // Initialize test environment
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        const accounts = await getIexecAccounts();
+        ({
+            requester,
+            appProvider,
+            datasetProvider,
+            scheduler,
+            anyone,
+            worker1,
+            worker2,
+            worker3,
+            worker4,
+            worker5,
+        } = accounts);
+        iexecWrapper = new IexecWrapper(proxyAddress, accounts);
+        ({ appAddress, datasetAddress, workerpoolAddress } = await iexecWrapper.createAssets());
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        ordersActors = {
+            appOwner: appProvider,
+            datasetOwner: datasetProvider,
+            workerpoolOwner: scheduler,
+            requester: requester,
+        };
+        ordersAssets = {
+            app: appAddress,
+            dataset: datasetAddress,
+            workerpool: workerpoolAddress,
+        };
+        ordersPrices = {
+            app: appPrice,
+            dataset: datasetPrice,
+            workerpool: workerpoolPrice,
+        };
+    }
+
+    it('Task Lifecycle with BoT Replication and Error Handling', async function () {
+        const workers = [worker1, worker2, worker3, worker4, worker5];
+        // Create deal.
+        const volume = 3;
+        const tasksAndWorkers: {
+            [key: number]: {
+                worker: SignerWithAddress;
+                useEnclave: boolean;
+                result: string;
+            }[];
+        } = {
+            0: [
+                { worker: worker1, useEnclave: false, result: 'iExec BOT 0' },
+                { worker: worker2, useEnclave: false, result: 'iExec BOT 0' },
+            ],
+            1: [
+                { worker: worker2, useEnclave: true, result: 'iExec BOT 1' },
+                { worker: worker3, useEnclave: true, result: 'iExec BOT 1' },
+            ],
+            2: [
+                { worker: worker1, useEnclave: false, result: 'iExec BOT 2' },
+                { worker: worker3, useEnclave: false, result: '<timeout reached>' },
```

Comment:
```suggestion
                { worker: worker3, useEnclave: false, result: '<reveal timeout reached>' },
```
?
To make it more precise.

Reply:
Actually it's not releated to a reveal timeout. Since it's a bad contribution when the trust contribution is reached the reveal of bad result would failed

Reply:
Why not `<bad contribution>` then?

Reply:
Perfect, makes sense